server/src/main/java/org/opensearch/telemetry/tracing/TracerModule.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.TracerPlugin;
+import org.opensearch.telemetry.Telemetry;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * A module for loading classes for tracer
+ *
+ * @opensearch.internal
+ */
+public class TracerModule {
+
+    private static final String TRACER_TYPE_DEFAULT_KEY = "tracer.type.default";
```

Comment:
It looks to me we have a flaw in design here: the plugin allows to register multiple telemetry providers but we could use only one of them (and we need yet another setting to say which one). 

If we cannot support multiple tracers, the plugin should return the `Optional<Telemetry>` than, the user should install the plugin it wants to use (OTEL / Jaeger / Zipkin / X-Ray) and be done with that.

Reply:
Have changed the signature to `Optional<Telemetry>` from `Map<String, Supplier<Telemetry>>`. But we would need the setting `tracer.type` to allow users mention the plugin they want to use. We would always have one plugin(default one) as its in `module`. If user installs another plugin we would have two plugins, so we need to provide a way to users to specify which plugin to use

Reply:
>But we would need the setting tracer.type to allow users mention the plugin they want to use. 

No need, if plugin is installed, we just use it. 

> If user installs another plugin we would have two plugins, so we need to provide a way to users to specify which plugin to use

We have to fail the startup here saying there is only one `Telemetry` provider supported (if `Optional<Telemetry>` returns non-empty telemetry).

Reply:
Done